include/oneapi/dnnl/dnnl.hpp

```diff
@@ -4654,21 +4654,18 @@ struct primitive_desc_base : public handle<dnnl_primitive_desc_t> {
 
         const_dnnl_memory_desc_t cdesc = dnnl_primitive_desc_query_md(
                 get(), dnnl::convert_to_c(what), idx);
-        if (!cdesc) return memory::desc();
 
-        dnnl_memory_desc_t cloned_md = nullptr;
-        error::wrap_c_api(dnnl_memory_desc_clone(&cloned_md, cdesc),
-                "could not clone a memory descriptor");
+        auto nc_cdesc = const_cast<dnnl_memory_desc_t>(cdesc);
 
-        return memory::desc(cloned_md);
+        return nc_cdesc ? memory::desc(nc_cdesc) : memory::desc();
```

Comment:
I believe changing the API semantics in this way may be problematic. It implicitly binds the lifetime of the returned object to that of the primitive descriptor, which significantly complicates the API's usability. The code snippet below illustrates one way this can lead to issues.
```cpp
auto src_md = convolution_forward::primitive_desc(engine, ...).src_desc();
// src_md is no longer valid.
```
If we're aiming to reduce overhead, I'd recommend making the option to skip cloning opt-in, similar to how the `allow_empty` feature works.

Reply:
Or, as an alternative, count references.

Reply:
One more thing. `memory::desc` class is based on the `handle` class that releases the handle upon destruction so it seems that it will try to release `nc_cdesc`.